### PR TITLE
kubernetes: support custom cluster dns domain

### LINKED ISSUE #1280: chore: support custom trust domain
https://github.com/envoyproxy/gateway/blob/2e2111494294cb80f591e8d148a29ac84c23b634/internal/infrastructure/kubernetes/ratelimit_service.go#L50

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

**What this PR does / why we need it**:

kubernetes: support custom cluster dns domain

**Which issue(s) this PR fixes**:

Fixes #1280

### CODE REVIEW

**Review by arkodg — APPROVED** [review]
LGTM, thanks !

File: internal/xds/translator/translator_test.go [hunk 1]
```diff
@@ -244,7 +253,7 @@ func TestTranslateXdsWithExtension(t *testing.T) {
 			ir := requireXdsIRFromInputTestData(t, "extension-xds-ir", tc.name+".yaml")
 			tr := &Translator{
 				GlobalRateLimit: &GlobalRateLimitSettings{
-					ServiceURL: ratelimit.GetServiceURL("envoy-gateway-system"),
+					ServiceURL: ratelimit.GetServiceURL("envoy-gateway-system", "cluster.local"),
```
Comment (qicz) [inline internal/xds/translator/translator_test.go:256]: maybe use `DefaultDNSDomain` is better

File: internal/xds/translator/translator_test.go [hunk 2]
```diff
@@ -135,10 +140,14 @@ func TestTranslateXds(t *testing.T) {
 	for _, tc := range testCases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
+			dnsDomain := tc.dnsDomain
+			if dnsDomain == "" {
+				dnsDomain = "cluster.local"
```
Comment (qicz) [inline internal/xds/translator/translator_test.go:145]: maybe use `DefaultDNSDomain` is better

File: charts/gateway-helm/values.tmpl.yaml [hunk 3]
```diff
@@ -23,9 +23,9 @@ deployment:
         cpu: 5m
         memory: 64Mi
   ports:
-  - name: grpc
-    port: 18000
-    targetPort: 18000
+    - name: grpc
+      port: 18000
```
Comment (qicz) [inline charts/gateway-helm/values.tmpl.yaml:27]: looks the lint differently others

Comment (zirain) [reply charts/gateway-helm/values.tmpl.yaml:27]: format by vscode

File: internal/xds/translator/translator_test.go (hunk 1 above)
Comment (zirain) [reply internal/xds/translator/translator_test.go:256]: it's ok to a test.

**Review by qicz — APPROVED** [review]

**Pull request merged**